connection.go

```diff
@@ -18,27 +18,31 @@ import (
 	"math/rand"
 	"sync/atomic"
 	"time"
+	"unsafe"
 )
 
-const (
-	sDisconnected int32 = 5 + iota
-	sConnected
-)
+func (ae *Exporter) loadLastConnectError() *error {
```

Comment:
I think we should abstract these loads and stores to perhaps
```go
func (ae *Exporter) lastConnectError() error {
    ptr := (*error)(atomic.LoadPointer(&ae.saveLastConnectErr)
    if ptr == nil {
         return nil
    }
    return *ptr
}

func (ae *Exporter) saveLastConnectError(err error) {
    var errPtr *error
    if err != nil {
        errPtr = &err
    }
    atomic.StorePointer(&ae.saveLastConnectErr, unsafe.Pointer(errPtr))
}
```

and then use them normally in our code instead of dereferences after checking if nil or not on error pointers.

Reply:
Good idea.